### DISCUSSION TITLE
How to implement arithmetic logic

### DISCUSSION THREAD

**Discussion by satish-srinivasa** [react-querybuilder/react-querybuilder#410] [General]
Hi,
        The react-querybuilder is an awesome component. I would like to know if this component supports arithmetic logic like Addition, Subtraction, division, count etc.

ex, select bill_amt + tax from sales or select * from employee where count(employee_id) >1

**Comment by jakeboone02** [comment]
Hi @satish-srinivasa. This library is designed to handle a `WHERE` clause, not a `SELECT` clause, so your first example of `select bill_amt + tax from sales` would not be covered.

However your second example, `select * from employee where count(employee_id) >1`, could theoretically be covered, but it would take some customization. RQB doesn't really come with out-of-the-box support for functions or arithmetic operations, even on the "value" (right-hand) side, and definitely not on the "field" (left-hand) side.

Something like this might point you in the right direction, though ([StackBlitz link here](https://stackblitz.com/edit/github-q8ems4-ajuncc?file=src/App.tsx)):

```tsx
const fns = ['count', 'min', 'max'];

const fields = ['age', 'weight', 'grade'];

const MyFieldSelector = ({ value, handleOnChange }: FieldSelectorProps) => {
  const [func, field] = value.split('|');

  return (
    <div>
      <select
        value={func}
        onChange={e => handleOnChange([e.target.value, field].join('|'))}>
        {fns.map(fn => (
          <option key={fn} value={fn}>
            {fn.toUpperCase()}
          </option>
        ))}
      </select>
      (
      <select
        value={field}
        onChange={e => handleOnChange([func, e.target.value].join('|'))}>
        {fields.map(fld => (
          <option key={fld} value={fld}>
            {fld.toUpperCase()}
          </option>
        ))}
      </select>
      )
    </div>
  );
};

export const App = () => {
  const [query, setQuery] = useState(initialQuery);

  return (
    <div>
      <QueryBuilder
        query={query}
        onQueryChange={q => setQuery(q)}
        controlElements={{
          fieldSelector: MyFieldSelector,
        }}
        getInputType={() => 'number'}
      />

      <h4>Query</h4>
      <pre>
        <code>{formatQuery(query, 'json')}</code>
      </pre>
    </div>
  );
};
```
